src/main/java/com/gruzini/tennistico/controllers/JoinMatchController.java

```diff
@@ -22,6 +25,8 @@ public JoinMatchController(JoinMatchService joinMatchService) {
 
     @GetMapping("/{match_id}")
     public String joinMatch(@PathVariable(name = "match_id") final Long matchId, final Principal principal) {
+        if (isNull(matchId)) throw new ResourceNotFoundException();
+
```

Comment:
chcemy rzucać exceptiona od razu w controllerze?
serwis i tak rzuci, jak będziemy procesować nieistniejący matchId
i obsłużymy wtedy w @ControllerAdvice handlerze

Reply:
no niby jo, nie pamiętałem że tam gdzieś są jakieś exceptiony rzucane niżej

Reply:
Ale chyba jednak nie ten match id to requestParam. Zobacz że nigdy nie jest po nim nic szukane. Jest on tylko dodawany do modelu. Jeżeli będzie nullem to wychrzani templateResolveException. Może nazwa exceptiona nie jest najfortunniejsza to się zgadzam